java-checks/src/main/java/org/sonar/java/checks/ProtectedMemberInFinalClassCheck.java

```diff
@@ -47,29 +52,39 @@
   public void visitNode(Tree tree) {
     ClassTree classTree = (ClassTree) tree;
     if (ModifiersUtils.hasModifier(classTree.modifiers(), Modifier.FINAL)) {
-      for (Tree member : classTree.members()) {
-        checkMember(member);
-      }
+      classTree.members().forEach(this::checkMember);
     }
   }
 
   private void checkMember(Tree member) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Private method 'checkMember' is never used. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AUnusedPrivateMethod)